src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
```

Comment:
PullProofAI

The guard implementation looks solid. It correctly checks the presence and validity of the feedback secret in the request headers. This is a good security practice to ensure that only authorized requests are allowed.

Overall, no issues found in the `FeedbackSecretGuard` implementation. It follows the standard NestJS guard pattern and effectively handles unauthorized requests based on the feedback secret provided in the headers.


<!-- This is an auto-generated comment by PullProofAI -->